odoo/addons/base/res/res_users.py

```diff
@@ -510,6 +514,36 @@ def check(cls, db, uid, passwd):
         finally:
             cr.close()
 
+    def _get_session_token_fields(self):
+        token_fields = set(USER_PRIVATE_FIELDS + ['active', 'password_crypt'])
+        return set(self.fields_get_keys()).intersection(token_fields)
+
+    @tools.ormcache('uid', 'sid')
+    def _compute_session_token(self, sid, uid):
+        """ Compute a session token given a session id and a user id """
+        # retrieve user's password crypt
+        session_fields = ', '.join(self._get_session_token_fields()) # SQL-i prone
```

Comment:
It should be safe to sql-i.
But you can quote identifier using [quote_ident](http://initd.org/psycopg/docs/extensions.html#psycopg2.extensions.quote_ident) or using [psycopg2.sql](http://initd.org/psycopg/docs/sql.html)